.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,106 @@
+name: Publish docs
+
+on:
+  push:
+    branches: [master]
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  dump-context:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Dump GitHub context
+        env:
+          GITHUB_CONTEXT: ${{ toJson(github) }}
+        run: echo "$GITHUB_CONTEXT"
+
+  get-changes-scope:
+    runs-on: ubuntu-latest
+    outputs:
+      changed-packages: ${{ steps.map-changed-files-to-pkg.outputs.result }}
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get PR for push event HEAD commit
+        uses: ./.github/actions/get-pr-for-commit
+        id: get-pr-for-commit
+        with:
+          commit-sha: ${{ github.sha }}
+
+      - name: List changed files
+        uses: ./.github/actions/list-changed-files
+        id: list-changed-files
+        with:
+          pull-number: ${{ steps.get-pr-for-commit.outputs.pull-number }}
+
+      - name: Get scope of changed packages
+        uses: actions/github-script@v4
+        id: map-changed-files-to-pkg
+        with:
+          script: |
+            const files = ${{ steps.list-changed-files.outputs.changed-files }}
+            // yarn doc flow only generates docs for JS
+            const regexp = /[a-zA-Z\-]+\/(js)/g
```

Comment:
Confused about this regex as the files that'll be updated will be TypeScript, i.e. `*.ts` files.

Reply:
Yeah, I previously used this regex to identify packages - which were in the form `<pkg>/js` or `<pkg>/program`. But, I think regex matching on `*.ts` should work too.

Reply:
Ah ok so this matches on the folder .. got it.